src/main/java/com/google/sps/data/Challenge.java

```diff
@@ -0,0 +1,37 @@
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Challenge {
+  private String challenge_type;
+  private List <String> steps;
+
+  public Challenge(String type, String name, List steps){  
+    this.challenge_type = type;
+    this.steps = new ArrayList <>(steps);
+  }
+
+ /* public boolean isValidChallenge(){
+  
+  } */
+
+  public boolean isEqual(Challenge object) {
+    if (this == object){
+      return true;
+    }else if (object instanceof Challenge){
+
+      Challenge otherChallenge = (Challenge) object;
+      if (challenge_type.equals(object.challenge_type) &&
```

Comment:
^ along with above, "name" doesn't seem to have any role in comparing challenges, is this intentional?

Reply:
No, I forgot to add a name parameter. Will fix. Thank you.

Reply:
I don't see your fix to address Natalie's comment. Please don't forget 🙏🙏!